service/internal/checksum/middleware_checksum_metrics_tracking_test.go

```diff
@@ -0,0 +1,169 @@
+//go:build go1.16
```

Comment:
1. Why do we need to duplicate the user agent test? If we just want to test that it includes the features we're interested in, I wouldn't duplicate the whole test but just check that the user agent includes the features we're interested in
2. Why do we put this under `go:build` tags?
3. If we need to do the above, can we change this to go 1.21?